build-support/virtualenv

```diff
@@ -9,8 +9,21 @@ VIRTUALENV_PACKAGE_LOCATION=${VIRTUALENV_PACKAGE_LOCATION:-https://pypi.io/packa
 REPO_ROOT=$(cd $(dirname "${BASH_SOURCE[0]}") && git rev-parse --show-toplevel)
 source ${REPO_ROOT}/build-support/common.sh
 
+function determine_python_bin_name() {
+  if [ "${PANTS_USE_PYTHON3_UNCONSTRAINED}" = true ]; then
+    py_version="3"
+  elif [ "${PANTS_USE_PYTHON36}" = true ]; then
+    py_version="3.6"
+  elif [ "${PANTS_USE_PYTHON37}" = true ]; then
+    py_version="3.7"
+  else
+    py_version="2.7"
+  fi
+  echo "python${py_version}"
+}
+
 if [[ -z "${PY}" ]]; then
```

Comment:
Why all the env vars? PY has existed for quite some time. Can't the caller just say PY=python{2.7,3.{6,7,8,...}} - this script should know nothing of the artificial 3.7 limit for example - it should continue to work when we support 3.8, etc.